content/blog/nats-server-29-release.md

```diff
@@ -0,0 +1,489 @@
+
+A related feature is a new configuration option in the server called [`max_outstanding_catchup`][max_outstanding_catchup] which limits the total bytes that are *in-flight* during stream catch-up. This was introduced to control how much bandwidth should be dedicated during catch-up to guard against saturating and degrading performance of the network.
+
+```config
+jetstream {
+  max_outstanding_catchup: 32MB
+}
+```
+
+[max_outstanding_catchup]: https://docs.nats.io/running-a-nats-service/configuration#jetstream
+
+#### Improved message distribution for multi-subscription pull consumers
+
+Given two (or more) subscriptions bound to a pull consumers, each subscription sends a _fetch_ request requesting a batch of messages from the server. Prior to 2.9, the requests were serialized and each one would get the full batch of requested messages if available.
+
+<img src="/img/nats-server-29-release/pull-fetch-prior29.jpg" style="display: block; margin: 0 auto 20px auto" width="500px" />
+
+This could result in an imbalanced distribution of messages across available subscriptions. Depending on the processing time of each message, this could significantly decrease throughput.
+
+In 2.9, the distribution algorithm was tweaked to fairly distribute messages across pending fetch requests.
+
+<img src="/img/nats-server-29-release/pull-fetch-29.jpg" style="display: block; margin: 0 auto 20px auto" width="500px" />
+
+For a set of fetch requests queued on the server, messages will be round-robin distributed until each fetch buffer is full or there are no more pending messages.
+
+#### Inactive threshold for durable consumers
+
+Consumers can be either ephemeral or durable. By design, an ephemeral consumer is intended to be used for one-off reads/replay of a stream and has a limited lifetime. In the client libraries, ephemerals are often created using one of the *convenience* methods. For example, in Go, using the [`JetStream`](https://pkg.go.dev/github.com/nats-io/nats.go#JetStream) interface.
+
+```go
+sub, _ := js.SubscribeSync("events.*")
+// Do some work...
+sub.Unsubscribe()
+```
+
+The client library handles:
+
+- Looking up the stream by pattern matching the subject being subscribed to
+- Auto-creating an *unnamed* consumer (which makes an ephemeral)
+- Initializing a subscription and binding it to the consumer
+
+From the server's persective, it keeps track of whether at least one subscription is bound to this ephemeral consumer _and_ if the subscription is still _active_. If/when it detects the subscription is no longer active, it will automatically delete the ephemeral consumer after the [`InactiveThreshold`](https://docs.nats.io/nats-concepts/jetstream/consumers#inactivethreshold) period (which defaults to five seconds).
```

Comment:
```suggestion
From the server's perspective, it keeps track of whether at least one subscription is bound to this ephemeral consumer _and_ if the subscription is still _active_. If/when it detects the subscription is no longer active, it will automatically delete the ephemeral consumer after the [`InactiveThreshold`](https://docs.nats.io/nats-concepts/jetstream/consumers#inactivethreshold) period (which defaults to five seconds).
```